<launch>
    <!-- Subscribed topics -->
    <param name="LiDAR_pointcloud_topic" type="string" value= "/laser_cloud_flat" />
    <!-- <param name="IMU_topic" type="string" value= "/livox/imu" /> -->
    <param name="IMU_topic" type="string" value= "/camera/imu" />
    <!-- <param name="Image_topic" type="string" value= "/camera/image_color" /> -->
    <param name="Image_topic" type="string" value= "/camera/color/image_raw" />
    <param name="ilive_common/map_output_dir" type="string" value="$(env HOME)/r3live_output" />
    <rosparam command="load" file="$(find ilive)/config/ilive_config.yaml" />

    <node pkg="ilive" type="ilive_LiDAR_front_end" name="ilive_LiDAR_front_end"  output="screen" required="true"/>
    <node pkg="ilive" type="ilive_mapping" name="ilive_mapping" output="screen" required="true" />
    
    <arg name="rviz" default="1" />
    <group if="$(arg rviz)">
        <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find ilive)/config/rviz/ilive_rviz_config.rviz" />
    </group>
</launch>
